codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -669,20 +703,36 @@ class HttpBindingGenerator(
                     isMultiValuedHeader = isMultiValuedHeader,
                 )
             val safeName = safeName("formatted")
-            rustTemplate(
-                """
-                let $safeName = $formatted;
-                if !$safeName.is_empty() {
-                    let header_value = $safeName;
+
+            if (serializeEmptyHeaders) {
+                rustTemplate(
+                    """
+                    let header_value = $formatted;
                     let header_value: #{HeaderValue} = header_value.parse().map_err(|err| {
                         #{invalid_field_error:W}
                     })?;
                     builder = builder.header("$headerName", header_value);
-                }
-                """,
-                "HeaderValue" to RuntimeType.Http.resolve("HeaderValue"),
-                "invalid_field_error" to renderErrorMessage("header_value"),
-            )
+
+                    """,
+                    "HeaderValue" to RuntimeType.Http.resolve("HeaderValue"),
+                    "invalid_field_error" to renderErrorMessage("header_value"),
+                )
+            } else {
```

Comment:
We might be able to use [conditionalBlock](https://github.com/smithy-lang/smithy-rs/blob/42751e5dbf4d51c06c085e4193bf013a7333a6f5/codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt#L125-L137) to consolidate these branches (may need to hoist `let $safeName = $formatted;`)

Reply:
Good call, did not know about `conditionalBlock`, updated both of the conditional sections in HttpBindingGenerator.kt to use it in https://github.com/smithy-lang/smithy-rs/pull/3887/commits/c2ecd75c7ad61b893e44082dfe9faf91d6b93693